typings/declarations.d.ts

```diff
@@ -1,36 +1,71 @@
 declare module "yoffee" {
-
     /**
      * Returns a `DocumentFragment` that has a one-way binding to each of the objects.
-     * @param {...Object} propsObjects Holds the state of this yoffee element
-     * @return a document fragment
+     * @param {object} propsObjects Objects which are used in the expressions inside the html for updating the html.
+     * @returns {(strings: string[], ...expressionCbs: Function[]) => DocumentFragment} html bound to `propsObjects` as a DocumentFragment.
      */
-    function html(...propsObjects: object[]): any; 
+    function html(...propsObjects: object[]): (strings: TemplateStringsArray, ...expressionCbs: Function[]) => DocumentFragment;
+
+    interface YoffeeElementWrapper extends YoffeeElement {
+        /**
+         * Called each time a property is updated. Attributes are converted to properties, so it is also called when an attribute changes.
+         * @param {string} prop The property that changed. Value can be accessed with `props[prop]`
+         */
+        onPropUpdate: (prop: string) => void
+
+        /**
+         * Called when the element is connected to the DOM after the element is created
+         */
+        onConnect: Function
+
+        /**
+         * Called when the element is disconnected from the DOM.
+         */
+        onDisconnect: Function
+    }
 
     /**
-     * 
-     * @param elementName a name for this html web component element
-     * @param element an arbitrary `Function`, or a (child of) `YoffeeElement`
-     * @throws an error if the second parameter isn't either a `YoffeeElement` subclass or a `Function`
+     * Creates a new Yoffee element, as a custom web component - a new html tag.
+     * @param elementName The new elements' name. Must include comma.
+     * @param element Declaration of the element: Either a callback function that returns the html, or a `YoffeeElement`
      */
-    function createYoffeeElement(elementName: string, element: Function | YoffeeElement): void;
-    
+    function createYoffeeElement(
+        elementName: string,
+        element: ((props: object, element: YoffeeElementWrapper) => DocumentFragment) | YoffeeElement
```

Comment:
I thought it took a function or element, good to know. and this change made it stronger typed :)

Reply:
> I thought it took a function or element

It still does, just defined more strongly as you said :)